kafka/openconfig/encoder.go

```diff
@@ -55,19 +54,15 @@ func NewEncoder(topic string, key sarama.Encoder, dataset string) kafka.MessageE
 	}
 }
 
-func (e *elasticsearchMessageEncoder) Encode(message proto.Message) ([]*sarama.ProducerMessage,
+func (e *elasticsearchMessageEncoder) Encode(response *pb.SubscribeResponse) (
+	[]*sarama.ProducerMessage,
 	error) {
 
-	response, ok := message.(*pb.SubscribeResponse)
-	if !ok {
-		return nil, UnhandledMessageError{message: message}
-	}
 	update := response.GetUpdate()
 	if update == nil {
 		return nil, UnhandledSubscribeResponseError{response: response}
 	}
-	updateMap, err := openconfig.NotificationToMap(e.dataset, update,
-		elasticsearch.EscapeFieldName)
+	updateMap, err := gnmi.NotificationToMap(update)
```

Comment:
> I don't know if you tested this but gnmi.NotificationToMap() isn't really a drop in replacement for openconfig.NotificationToMap().

Yes but also, gnmi.Path is not a drop in replacement for openconfig.Path. :-)

> I think eventually we'll end up flattening the schema and consolidating it with what we have in our own backend, but in the meanwhile would you be up for keeping the format the same? If 

@7AC may I ask why we really need to?  I see [`cmd/ocsplunk`](https://github.com/aristanetworks/goarista/blob/master/cmd/ocsplunk/main.go#L95) uses the flattened path value of gnmi.NotificationToMap() directly for writing events, which I believe is because that's how it's supposed to be. Doesn't the same go for kafka agent?

Reply:
The initial use case for ockafka was for Kafka -> Logstash -> Elasticsearch -> Kibana. Having an entire device as a single JSON document makes a really nice representation in Elasticsearch, and that's what you see implemented here (before your changes). It does run into some limitations when it comes to scaling (max number of fields, fields changing type, etc). We have some changes coming to ockafka that work around these limitations which will cause schema changes. As a result I'm a bit reluctant to make an additional schema change here. So I think we can go two routes:
* hold off merging this and can reconcile with the changes on our end once they come in
* change the PR so the schema we output looks the same (basically https://github.com/aristanetworks/goarista/pull/22#discussion_r234832013)

What do you think? By the way sorry for the delayed response it was a long weekend in the US.

Reply:
@7AC many thanks for the clarification. Now I see how you plan to have it grow, the route 1 sounds natural to me. 
> hold off merging this and reconcile with the changes on our end once they come in

Hope your revision for the work around will go smoothly. And also hope you enjoyed the holidays!